### PR TITLE
Fix build issues

### PR DESCRIPTION
**Pull request by msdmkats**
This PR fixes 2 problems:
1. CMake crashes in VS 2017/2019
2. Inconsistent output paths between generators - #19 https://github.com/awf/ADBench/issues/26#issuecomment-496293313

Solutions:
1. Update HunterGate (after that VS 2019 consistently works w/o errors on at least 2 machines, before it consistently crashed)
2. Set 
CMAKE_*_OUTPUT_DIRECTORY to 
generator expressions resolving to empty values in order 
to avoid creation of configuration related folders by multi-config generators such as `Visual Studio X X` https://cmake.org/cmake/help/v3.5/prop_tgt/RUNTIME_OUTPUT_DIRECTORY.html#prop_tgt:RUNTIME_OUTPUT_DIRECTORY

I also added CMakeSettings.json with sensible default config that puts CMakeCache into a folder nearby and makes x64-Release the default configuration for Visual Studio.

### CODE REVIEW

**Comment by msdmkats** [comment]
Sorry, had to recreate it. Apparently, PRs cant be reopened after a force-push.

**Comment by msdmkats** [comment]
Regarding https://github.com/awf/ADBench/pull/38#pullrequestreview-242737266
> 1. `CMakeSettings.json` was added and then removed.  Could you then remove those two commits from the PR?
> 2. Is it possible to pull in `HunterGate.cmake` from an upstream source rather than including it verbatim in our repo?

1. I decided to put CMakeSettings.json back - without it VS 2017 builds the cache indefinitely on my machine.
2. Probably, yes - from here: https://github.com/hunter-packages/gate
I put a copy, because their doc https://docs.hunter.sh/en/latest/quick-start/boost-components.html suggested it.

**Comment by tomjaguarpaw** [comment]
1. OK, seems reasonable!

2. Good idea to follow the upstream authors suggestion.

**Comment by tomjaguarpaw** [comment]
I am happy with this but @awf needs to see it to check that he's happy with the output file build location.  See https://github.com/awf/ADBench/pull/37#discussion_r288160629.

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,28 +4,36 @@
...
 else ()
-	set(AD_BUILD_TYPE "Release")
+    set(AD_BUILD_TYPE "Release")
 endif ()
 
+# $<1:> always resolves to empty string, but the fact, that a generator
+# expression is used, forces Visual Studio generator to not create
+# configuration-specific folders for outputs, thus making output paths
+# consistent between different generators.
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY $<1:>)
```
Comment (tomjaguarpaw) [inline CMakeLists.txt]: via @awf:  Add comment here, linking to CMake docs as in pr.

Comment (tomjaguarpaw) [reply CMakeLists.txt]: (Given that there's a comment on this version I suppose a link to the CMake docs will suffice)

Comment (msdmkats) [reply CMakeLists.txt]: Done (amended the last commit, diff here didn't update for me, but the link is there now).

Comment (tomjaguarpaw) [reply CMakeLists.txt]: That looks good to me.  Probably best to wait for final @awf approval.

**Comment by awf** [comment]
lgtm.  Please merge and test clean checkout from master.

File: CMakeLists.txt (hunk 1 above)
Comment (msdmkats) [reply CMakeLists.txt]: Got that approval below.

**Comment by msdmkats** [comment]
Done. Tested on VS 2017, VS 2019, and console builds with Ninja and "Visual Studio 2017 Win64" generators on Windows 10. Works.

**Comment by tomjaguarpaw** [comment]
Excellent

**Comment by tomjaguarpaw** [comment]
@msdmkats Please delete the branch if you no longer need it.

**Pull request merged**